features/expo/deviceInfo/hooks.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+
+import { enumToStr } from 'htk/utils/enum';
+import * as Device from 'expo-device';
+import { DateTime } from 'luxon';
+
+import { DeviceInfo } from './constants';
+import { DeviceType } from './enums';
+
+export const useDeviceInfo = () => {
+    const [deviceInfo, setDeviceInfo] = useState<DeviceInfo>({
+        brand: null,
+        manufacturer: null,
+        deviceName: null,
+        modelId: null,
+        modelName: null,
+        osName: null,
+        osVersion: null,
+        osBuildId: null,
+        osInternalBuildId: null,
+        deviceType: null,
+        deviceYearClass: null,
+        isDevice: null,
+        supportedCpuArchitectures: null,
+        totalMemory: null,
+        uptime: null,
+        isJailBroken: null,
+    });
+
+    useEffect(() => {
+        const fetchDeviceInfo = () => {
+            Promise.all([
+                Device.getDeviceTypeAsync(),
+                Device.getUptimeAsync(),
+                Device.isRootedExperimentalAsync(),
+            ])
```

Comment:
Please use separate promises. This will fail on all, even if only one promise fails.